### ISSUE TITLE
Enable loading TF models serialized with SavedModel format

### ISSUE THREAD

**Issue by edgimar** [NVIDIA/DIGITS#1775]
There should be a way to load a TensorFlow model that has been serialized as a [SavedModel](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/saved_model/README.md) and use it as a model within DIGITS (for both training and inference).

Ideally, an interface might at least have the following options:

- path to SavedModel
- name of signature to use
- signature keys of input placeholders (e.g. x, y, is_training)
- signautre keys of output variables (e.g. cost and logits)

I'm probably missing details of what else would be required, but this would be a good start.

If I could accomplish this by modifying the `model.py` file, I would, but it is difficult if not impossible to replace the input placeholder nodes in a loaded SavedModel graph with the x and y attributes that DIGITS defines in the Tower class).  If there were a way for Tower to have methods like get_x_placeholder(), then these could be overridden in subclasses, allowing them to determine which TF tensor to use as the placeholder.

**Comment by ethantang95** [comment]
It should be possible if you redefine it as a model property. Currently when you write tensorflow models for DIGITS, the inference and loss function both acts as properties. If possible, you can then redefine X and Y to be model properties and have that function in the background

**Comment by edgimar** [comment]
@ethantang95 Do you mean by this that it's currently possible (if so, can you provide a minimal example that demonstrates how), or do you mean that it would be a way forward if such a capability were to be implemented?
